packages/m2m-gateway/src/utils/polling.ts

```diff
@@ -25,6 +27,36 @@ export function pollResourceWithMetadata<T>(
   });
 }
 
+/**
+ * Polls a resource by repeatedly calling the provided fetch function until the resource is deleted
+ * (i.e., a 404 Not Found is returned) or the maximum number of retries is exceeded.
+ *
+ * @param fetch - A function that returns a Promise resolving when the resource exists,
+ *                and rejects with a 404 error when the resource is deleted.
+ * @returns  Resolves when the resource is confirmed deleted.
+ */
+export async function pollResourceUntilDeletion(
```

Comment:
Consider moving this in [apiUtils.ts](https://vscode.dev/github/pagopa/interop-be-monorepo/blob/develop/packages/commons/src/utils/apiUtils.ts#L15) in commons, together with the `createPollingByCondition` utility - maybe it could become `createPollingUntilDeletion`